src/frog-base.tsx

```diff
@@ -186,7 +185,64 @@ export type RouteOptions<method extends string = string> = Pick<
   (method extends 'frame' | 'image'
     ? {
         fonts?: ImageOptions['fonts'] | (() => Promise<ImageOptions['fonts']>)
-      }
+      } & (method extends 'frame'
+        ? {
+            preview?: {
+              /**
+               * Path of the next frame.
+               *
+               * @example '/submit'
+               */
+              action?: FrameResponse['action']
+              /**
+               * HTTP response headers.
+               */
+              headers?: FrameResponse['headers']
+              /**
+               * An icon ID.
+               *
+               * @see https://warpcast.notion.site/Spec-Farcaster-Actions-84d5a85d479a43139ea883f6823d8caa
+               * @example `'log'`
+               */
+              icon?: Octicon
+              /**
+               * Title of the frame (added as `og:title`).
+               *
+               * @example 'Hello Frog'
+               */
+              title?: FrameResponse['title']
+            } & (
+              | {
+                  /**
+                   * Preview Image to render for the frame. Can either be a JSX element, or URL.
+                   *
+                   * @example
+                   * <div style={{ fontSize: 60 }}>Hello Frog</div>
+                   */
+                  image?: JSX.Element
+                  /**
+                   * Image options.
+                   *
+                   * @see https://vercel.com/docs/functions/og-image-generation/og-image-api
+                   *
+                   * @example
+                   * { width: 1200, height: 630 }
+                   */
+                  imageOptions?: Pretty<Omit<ImageOptions, 'fonts'>> | undefined
+                }
+              | {
+                  /**
+                   * Preview Image to render for the frame. Can either be a JSX element, or URL.
+                   *
+                   * @example
+                   * "https://i.ytimg.com/vi/R3UACX5eULI/maxresdefault.jpg"
+                   */
+                  image?: string
+                  imageOptions?: never
+                }
+            )
```

Comment:
cannot do a `Pick<FrameResponse, 'image' | 'imageOptions'>` as it breaks discriminated union.